File: 0023-merge-k-sorted-lists/0023-merge-k-sorted-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeKLists(vector<ListNode*>& lists) {
        
            
            vector<int>ans;
            
            for(int i=0;i<lists.size();i++){
                    
                
                        while(lists[i]!=NULL){
                                
                                ans.push_back(lists[i]->val);
                                lists[i]=lists[i]->next;
                        }
                        
                
            }
           
            sort(ans.begin(),ans.end());
        
            ListNode* temp=new ListNode();
            ListNode* a=temp;
            for(int i=0;i<ans.size();i++){
                    temp->next=new ListNode(ans[i]);
                    temp=temp->next;
                
            }
        
        
            return a->next;
    }
};

File: 0023-merge-k-sorted-lists/0023-merge-k-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeKLists(self, lists: List[Optional[ListNode]]) -> Optional[ListNode]:
        
            ans=[]
            for item in lists:
                    temp=item
                    while(temp):
                            ans.append(temp.val)
                            temp=temp.next

            if not len(ans): return None
            ans.sort()
            print(ans)       
            temp=ListNode()
            a=temp
            for item in ans:
                    
                    temp.next=ListNode(item)
                    temp=temp.next
                    
                    
            return a.next
                            
                        
                        
                    
        